include/openthread/platform/ble.h

```diff
@@ -0,0 +1,1013 @@
+ * @param[out]  aMtu       A pointer contains current ATT_MTU value.
+ *
+ * @retval ::OT_ERROR_NONE     ATT_MTU value has been placed in @p aMtu.
+ * @retval ::OT_ERROR_FAILED   BLE Device cannot determine its ATT_MTU.
+ */
+otError otPlatBleGattMtuGet(otInstance *aInstance, uint16_t *aMtu);
+
+/*******************************************************************************
+ * @section Bluetooth Low Energy GATT Client.
+ ******************************************************************************/
+
+/**
+ * Sends ATT Read Request.
+ *
+ * @note This function shall be used only for GATT Client.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ * @param[in] aHandle    The handle of the attribute to be read.
+ *
+ * @retval ::OT_ERROR_NONE           ATT Write Request has been sent.
+ * @retval ::OT_ERROR_INVALID_STATE  BLE Device is in invalid state.
+ * @retval ::OT_ERROR_INVALID_ARGS   Invalid handle value has been supplied.
+ * @retval ::OT_ERROR_NO_BUFS        No available internal buffer found.
+ */
+otError otPlatBleGattClientRead(otInstance *aInstance, uint16_t aHandle);
+
+/**
+ * The BLE driver calls this method to notify OpenThread that ATT Read Response
+ * packet has been received.
+ *
+ * This method is called only if @p otPlatBleGattClientRead was previously requested.
+ *
+ * @note This function shall be used only for GATT Client.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ * @param[in] aPacket    A pointer to the packet contains read value.
+ *
+ */
+extern void otPlatBleGattClientOnReadResponse(otInstance *aInstance, otBleRadioPacket *aPacket);
+
+/**
+ * Sends ATT Write Request.
+ *
+ * @note This function shall be used only for GATT Client.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ * @param[in] aHandle    The handle of the attribute to be written.
+ * @param[in] aPacket    A pointer to the packet contains value to be
+ *                       written to the attribute.
+ *
+ * @retval ::OT_ERROR_NONE           ATT Write Request has been sent.
+ * @retval ::OT_ERROR_INVALID_STATE  BLE Device is in invalid state.
+ * @retval ::OT_ERROR_INVALID_ARGS   Invalid handle value, data or data length has been supplied.
+ * @retval ::OT_ERROR_NO_BUFS        No available internal buffer found.
+ */
+otError otPlatBleGattClientWrite(otInstance *aInstance, uint16_t aHandle, otBleRadioPacket *aPacket);
+
+/**
+ * The BLE driver calls this method to notify OpenThread that ATT Write Response
+ * packet has been received.
+ *
+ * This method is called only if @p otPlatBleGattClientWrite was previously requested.
+ *
+ * @note This function shall be used only for GATT Client.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ * @param[in] aHandle    The handle on which ATT Write Response has been sent.
+ *
+ */
+extern void otPlatBleGattClientOnWriteResponse(otInstance *aInstance, uint16_t aHandle);
+/**
+ * Subscribes to characteristic.
```

Comment:
The API has been created for specific application, where subscribing means "subscribe for indications" (not for notifications). I guess we should clearly state it in the documentation or otherwise we should extend this API to allow user to choose - but i guess it may be done as an improvement later on.